src/fiber/fiber.mli

```diff
@@ -227,6 +227,29 @@ module Mvar : sig
   val write : 'a t -> 'a -> unit fiber
 end
 
+module Svar : sig
+  (** A reference where:
+
+      - writes are always immediate
+      - reads can provide a predicate *)
+
+  (** ['a t] is a reference of value of type ['a] *)
+  type 'a t
+
+  (** [create init] creates a new svar initialized with [init] *)
+  val create : 'a -> 'a t
+
+  (** [read t] returns the current value in [t] *)
+  val read : 'a t -> 'a
+
+  (** [wait t ~until] read from [t] until the first value [v] that satisfies
+      [until v] *)
```

Comment:
```suggestion
  (** [wait t ~until] waits until [t]'s value satisfies the predicate [until].
      If the current value satisfies it already, [wait] returns immediately, 
      otherwise, the caller is blocked. *)
```